packages/macro/src/macroJs.ts

```diff
@@ -418,6 +446,31 @@ export default class MacroJs {
     )
   }
 
+  createMessageDescriptorFromIdMsg(
+    idTokens: Tokens,
+    msgTokens: Tokens,
+    oldLoc?: SourceLocation
+  ) {
+    const { message, values } = buildICUFromTokens(msgTokens)
+    const { message: id } = buildICUFromTokens(idTokens)
+
+    const properties: ObjectProperty[] = [
+      this.createObjectProperty(ID, this.types.stringLiteral(id)),
+
+      !this.stripNonEssentialProps
+        ? this.createObjectProperty(MESSAGE, this.types.stringLiteral(message))
+        : null,
+
+      this.createValuesProperty(values),
+    ]
+
+    return this.createMessageDescriptor(
+      properties,
+      // preserve line numbers for extractor
+      oldLoc
+    )
+  }
+
```

Comment:
Let's DRY it, and instead of creating a separate function, just extend `createMessageDescriptorFromTokens` by adding an optional `id?: Node`

```ts
  createMessageDescriptorFromTokens(tokens: Tokens, oldLoc?: SourceLocation, idNode?: Node) {
    const { message, values } = buildICUFromTokens(tokens)
    
        const properties: ObjectProperty[] = [
          id ? this.createObjectProperty(ID, idNode) : this.createIdProperty(message),
    
```